src/resource_clients/request_queue.ts

```diff
@@ -96,6 +98,108 @@ export class RequestQueueClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    protected async _batchAddRequestsWithRetries(
+        requests: Omit<RequestQueueClientRequestSchema, 'id'>[],
+        options: RequestQueueClientBatchAddRequestOptions = {},
+    ): Promise<RequestQueueClientBatchAddRequestsResult> {
+        // Keep track of the requests that remain to be processed (in parameter format)
+        let remainingRequests = requests;
+        // Keep track of the requests that have been processed (in api format)
+        let processedRequests: ProcessedRequest[] = [];
+        // The requests we have not been able to process in the last call
+        // ie. those we have not been able to process at all
+        let unprocessedRequests: UnprocessedRequest[] = [];
+        for (let i = 0; i < 1 + this.httpClient.maxRetries; i++) {
+            try {
+                const response = await this.httpClient.call({
+                    url: this._url('requests/batch'),
+                    method: 'POST',
+                    data: remainingRequests,
+                    params: this._params({
+                        forefront: options.forefront,
+                        clientKey: this.clientKey,
+                    }),
+                });
+                const data = response.data.data as RequestQueueClientBatchAddRequestsResult;
+                processedRequests = [...processedRequests, ...data.processedRequests];
```

Comment:
is there any reason to create new arrays here instead of just pushing to existing one? destructing may look nice, but it will eat more memory than necessary if we are fine with mutating the array, as you are creating new one and pushing all existing items to it first - things you don't with simple `push`

```suggestion
                processedRequests.push(...data.processedRequests);
```